src/devices/bus/heathzenith/h8/h8bus.cpp

```diff
@@ -0,0 +1,370 @@
+{
+}
+
+device_h8bus_p1_card_interface::~device_h8bus_p1_card_interface()
+{
+}
+
+device_h8bus_p2_card_interface::device_h8bus_p2_card_interface(const machine_config &mconfig, device_t &device) :
+	device_h8bus_card_interface(mconfig, device)
+{
+}
+
+device_h8bus_p2_card_interface::~device_h8bus_p2_card_interface()
+{
+}
+
+DEFINE_DEVICE_TYPE(H8BUS_SLOT, h8bus_slot_device, "h8bus_slot", "Heath H-8 slot")
+
+h8bus_slot_device::h8bus_slot_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: h8bus_slot_device(mconfig, H8BUS_SLOT, tag, owner, clock)
+{
+}
+
+h8bus_slot_device::h8bus_slot_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock)
+	: device_t(mconfig, type, tag, owner, clock)
+	, device_single_card_slot_interface(mconfig, *this)
+	, m_h8bus(*this, finder_base::DUMMY_TAG)
+	, m_h8bus_slottag(nullptr)
+{
+}
+
+void h8bus_slot_device::device_start()
+{
+}
+
+void h8bus_slot_device::device_resolve_objects()
+{
+	device_h8bus_card_interface *dev = get_card_device();
+
+	if (dev)
+	{
+		dev->set_h8bus_tag(m_h8bus.target(), m_h8bus_slottag);
+		unsigned slot_num = m_h8bus->add_h8bus_card(*dev);
+		dev->set_slot_num(slot_num);
+	}
+}
+
+
+DEFINE_DEVICE_TYPE(H8BUS, h8bus_device, "h8bus", "Heathkit H8 bus (Benton Harbor Bus)")
+
+h8bus_device::h8bus_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: h8bus_device(mconfig, H8BUS, tag, owner, clock)
+{
+}
+
+h8bus_device::h8bus_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock)
+	: device_t(mconfig, type, tag, owner, clock)
+	, device_memory_interface(mconfig, *this)
+	, m_mem_config("mem", ENDIANNESS_LITTLE, 16, 16, 0, address_map_constructor(FUNC(h8bus_device::mem_map), this))
+	, m_io_config("io", ENDIANNESS_LITTLE, 8, 8, 0, address_map_constructor(FUNC(h8bus_device::io_map), this))
+	, m_p1_card(nullptr)
+	, m_p2_card(nullptr)
+{
+}
+
+h8bus_device::~h8bus_device()
+{
+}
+
+void h8bus_device::device_start()
+{
+}
+
+void h8bus_device::device_reset()
+{
+	m_int1_slot_states = 0;
+	m_int2_slot_states = 0;
+	m_int3_slot_states = 0;
+	m_int4_slot_states = 0;
+	m_int5_slot_states = 0;
+	m_int6_slot_states = 0;
+	m_int7_slot_states = 0;
+	m_reset_slot_states = 0;
+	m_hold_slot_states = 0;
+	m_m1_slot_states = 0;
+}
+
+void h8bus_device::mem_map(address_map &map)
+{
+	map.unmap_value_high();
+}
+
+void h8bus_device::io_map(address_map &map)
+{
+	map.unmap_value_high();
+	map.global_mask(0xff);
+}
+
+device_memory_interface::space_config_vector h8bus_device::memory_space_config() const
+{
+	return space_config_vector { std::make_pair(AS_PROGRAM, &m_mem_config), std::make_pair(AS_IO, &m_io_config) };
+}
+
+u8 h8bus_device::add_h8bus_card(device_h8bus_card_interface &card)
+{
+	m_device_list.emplace_back(card);
+	char* endptr;
+	long slot_num = strtol(&card.m_h8bus_slottag[1], &endptr, 10) - 1;
```

Comment:
Not a fan of depending on specific tag values, and not sure it's necessary here. Have you considered counting interrupt assert/deassert instead of tracking state bitwise by slot? While it does mean each card needs to be careful to only output its interrupt state when it changes, my opinion is that it more closely represents what's really happening with a wired-or signal like this. That is, the state of the bus line only changes when the first card asserts or last card deasserts the line.

Reply:
This bit-wise approach for the interrupts is also what Vas suggested on his comment - https://github.com/mamedev/mame/pull/13560#discussion_r2054828750